sky/serve/core.py

```diff
@@ -130,6 +99,7 @@ def up(
         controller_exist = (
             global_user_state.get_cluster_from_name(controller_name)
             is not None)
+        requested_cloud = list(task.resources)[0].cloud
```

Comment:
We should only do the inference of controller cloud if all resources uses the same cloud